### PR TITLE
D8CORE-000: Limit options in viewfield react.

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Allow altering data

# Needed By (Date)
- Today

# Urgency
- High

# Steps to Test

1. Check out https://github.com/SU-SWS/react_paragraphs/pull/44
2. Check out this branch
3. Clear all caches
4. Add a news views paragraph to a basic page
5. Ensure only [cards, list] show up as options for the display.

### CODE REVIEW

File: stanford_news.module [hunk 1]
```diff
@@ -176,6 +176,30 @@ function stanford_news_field_widget_form_alter(&$element, FormStateInterface $fo
   }
 }
 
+/**
+ * Implements hook_react_paragraphs_getfieldinfo_post_alter().
+ */
+function stanford_news_react_paragraphs_getfieldinfo_post_alter($field_element, $field_config, &$info) {
+
+  // Only alter data for the news_views view field.
+  if ($field_config->get('field_name') !== "su_news_view_field") {
```
Comment (pookmish) [inline stanford_news.module:185]:
```suggestion
  if ($field_config->getName() !== "su_news_view_field") {
```

File: stanford_news.module [hunk 2]
```diff
@@ -176,6 +176,30 @@ function stanford_news_field_widget_form_alter(&$element, FormStateInterface $fo
   }
 }
 
+/**
+ * Implements hook_react_paragraphs_getfieldinfo_post_alter().
+ */
+function stanford_news_react_paragraphs_getfieldinfo_post_alter($field_element, $field_config, &$info) {
```
Comment (pookmish) [inline stanford_news.module:182]: add `FieldConfigInterface` argument suggestion

File: stanford_news.module [hunk 3]
```diff
@@ -176,6 +176,30 @@ function stanford_news_field_widget_form_alter(&$element, FormStateInterface $fo
...
+
+  // Only alter data for the news_views view field.
+  if ($field_config->get('field_name') !== "su_news_view_field") {
+    return;
+  }
+
+  // We want to remove a few displays from the options.
+  // See stanford_news_field_widget_form_alter for the same appraoch.
+  $deny_list = ['topics_list', 'term_block', 'vertical_teaser_term_list'];
+  $vals = array_column($info['displays']['stanford_news'], 'value');
```
Comment (pookmish) [inline stanford_news.module:192]: can't say i've seen that function til now

Comment (sherakama) [reply stanford_news.module:192]: Yeah, it wasn't on top of the mind at first but I often review the array functions when it comes to working with multidimensional arrays and it turned out to be pretty darned handy.

**Pull request merged**